src/related-items-and-comparisons/RelatedItemsList.jsx

```diff
@@ -1,12 +1,62 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import CardItem from './CardItem.jsx';
 
 const RelatedItemsList = (props) => {
-  return (
-    <div className="related-items-list">
-      {props.products.map((product) => <CardItem key={product.id} product={product} />)}
-    </div>
-  );
+  const allProducts = props.products;
+  const products = props.products;
+  const [currList, setCurrList] = useState(products);
+  const [currIndex, setCurrIndex] = useState(0);
+
+  // "useEffect is componentDidMount, componentDidUpdate, and componentWillUnmount combined"
+
+  const next = () => {
+    setCurrIndex(currIndex + 1);
+    setCurrList(products.slice(currIndex + 1)); //bandaid fix, still not sure why its not updating the array without the +1 here
```

Comment:
`setCurrIndex` doesn't change the value of this `currIndex`, it changes what `currIndex` will be the next time the component renders.

Reply:
Gotcha, I'll keep that in mind (also leaving a comment in my code so I can see that while I work)